lib/collection-comparator.c

```diff
@@ -79,12 +83,18 @@ collection_comparator_start(CollectionComparator *self)
 }
 
 void
-collection_comparator_add_value(CollectionComparator *self, const gchar *value)
+collection_comparator_add_value(CollectionComparator *self, const gchar *value, time_t mtime)
 {
   CollectionComparatorEntry *entry = g_hash_table_lookup(self->original_map, value);
+
   if (entry)
     {
       entry->flag = IN_BOTH;
+      if (mtime && (entry->mtime != mtime))
+        {
+          self->handle_modified_entry(entry->value, self->callback_data);
```

Comment:
As mtime is never zero (except when `g_stat` fails), this callback is called for each files that are already tracked. Which currently includes a memory allocation, full path calculation, and a deallocation, which is unnecessary from wildcard file source perspective. Wildcard filesource can work on thousands of files, and executing these per rescan is unnecessary work.

This problem persists even with my previous comment how we could eliminate mtime here.

To resolve this, either handle_modified_entry should be set optionally, and called here like:
```
if (self->handle_modified_entry)
  self->handle_modified_entry(entry->value, self->callback_data);
```
or collection comparator could remember which events it should handle. For this case a mode field could be added  to `CollectionComparator` (for example `guint32 events`, with flags `COLLECTION_COMPARATOR_EVENT_NEW`,` ..._DELETED`,` ..._MODIFIED`).

Probably this latter is easier to impement.